### DISCUSSION TITLE
如何提升PR的效率

### DISCUSSION THREAD

**Discussion by fcwalker** [gogs/gogs#6672] [中文讨论]
试用的最新的docker镜像安装的，在进行PR合并的时候，耗时较长，有没有什么设置可以提高PR合并的效率呢？

**Comment by unknwon** [comment]
目前的 PR 实现完全是基于 Git 操作的，如果不是巨型 PR 但是合并耗时过长的话，可能应该考虑多给点 CPU？

**Reply by fcwalker** [reply]
我是用docker跑的，PR期间看了下资源使用情况，内存，cpu，磁盘，IO都很正常。
另外：今天使用gogs/gogs:latest的镜像重新构建，发现一直报健康检查失败，看了下最新版本的镜像这几天升级了，麻烦大大看看呢

**Reply by unknwon** [reply]
>今天使用gogs/gogs:latest的镜像重新构建，发现一直报健康检查失败，

没有错误日志的话，我基本帮不上什么忙。。

**Comment by fcwalker** [comment]
![image](https://user-images.githubusercontent.com/27950366/146726791-6cb02bfd-9d54-475c-b258-d93ab48c122a.png)
抱歉，忘上图了

**Reply by unknwon** [reply]
OK, 你发的日志症状和这个用户最近上报的貌似是一样的。。他也是用的 Docker https://github.com/gogs/gogs/issues/3801#issuecomment-991959361

这个 panic 是从 xorm 里面来的。。有可能是 xorm 的版本问题。。我看看吧

**Reply by unknwon** [reply]
我测试了一下，又看了你的日志，日志里说了没有找到 custom 配置文件，说明这是一个全新的 Gogs 容器，healthcheck挂掉是正常的，因为还没有配置数据库。我测试了提交安装页面之后就不会再出现这个 PANIC 了。

**Reply by fcwalker** [reply]
在原有的环境（docker swarm）上升级，没有产生日志，所以单节点部署测试的。
> 日志里说了没有找到 custom 配置文件

是不是新版本必须要custom配置文件才可以通过healthcheck呢

**Reply by fcwalker** [reply]
compose配置如下（出错版本image写的是gogs/gogs:latest）：
![image](https://user-images.githubusercontent.com/27950366/146750130-3fce967f-cfdd-4c03-a2ed-d655ae5c3326.png)

**Reply by unknwon** [reply]
连不上数据库的状态下，healthcheck挂掉也是合情合理的吧。。就是panic难看了点 哈哈

**Reply by unknwon** [reply]
我现在不太明白你是要反馈什么问题了。。

**Reply by fcwalker** [reply]
重新描述下：
两个问题：
1、PR有点慢，这个问题，先不管了
2、镜像升级到最新无法启动：
用0.12版本能够正常部署、启动
镜像文件使用latest版本就无法启动了，会报unhealth
版本升级是不是有什么需要注意的地方？

**Reply by unknwon** [reply]
为什么unhealthy就无法启动了？docker不允许吗？

**Reply by fcwalker** [reply]
哈哈，很神奇，集群自动认定失败了，日志都没有。
等空了，我用单节点升级试试，谢谢大佬

**Reply by unknwon** [reply]
可能是因为panic给了500的缘故，我简单修了一下 https://github.com/gogs/gogs/pull/6687

**Reply by unknwon** [reply]
合并了，Docker 镜像估计会在一个小时后被 push 上去 https://github.com/gogs/gogs/runs/4581237595?check_suite_focus=true
